libraries/AP_GPS/AP_GPS_NMEA.h

```diff
@@ -67,11 +67,12 @@ class AP_GPS_NMEA : public AP_GPS_Backend
 
 private:
     /// Coding for the GPS sentences that the parser handles
-    enum _sentence_types {      //there are some more than 10 fields in some sentences , thus we have to increase these value.
+    enum _sentence_types : uint8_t {      //there are some more than 10 fields in some sentences , thus we have to increase these value.
         _GPS_SENTENCE_RMC = 32,
         _GPS_SENTENCE_GGA = 64,
         _GPS_SENTENCE_VTG = 96,
         _GPS_SENTENCE_HDT = 128,
+        _GPS_SENTENCE_PHD = 138, // extension for AllyStar GPS modules
```

Comment:
Should note the other indexes being used

Reply:
you mean the +1 etc? That is the (somewhat strange) pattern used in the NMEA driver.

Reply:
Yeah. Didn't realize other msgs do it too. I'll take a look. I was just thinking we should at least mention what the range is as a comment.

Reply:
it's a horrible pattern, but it's how it is done in this driver